### PR TITLE
Meta: AVA -> Vitest

### PR DESCRIPTION
**Pull request by fregante**
- Related to #5799 

For speed and lack of hacks everywhere. The API may be less great.

~~I used Jest’ codemods to do most of the changes~~ and then fixed a couple of AVA-only features/awkwardness

_Edit: I reverted the codemod and did a simple search and replace to preserve the assert-style tests; Jest prefers `expect()`, which doesn't support custom error messages_

# Before

```
❯ nr ava

(node:52859) ExperimentalWarning: --experimental-loader is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:52859) ExperimentalWarning: --experimental-loader is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:52859) ExperimentalWarning: --experimental-loader is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:52859) ExperimentalWarning: --experimental-loader is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:52859) ExperimentalWarning: --experimental-loader is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:52859) ExperimentalWarning: --experimental-loader is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:52859) ExperimentalWarning: --experimental-loader is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:52859) ExperimentalWarning: --experimental-loader is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:52859) ExperimentalWarning: --experimental-loader is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:52859) ExperimentalWarning: --experimental-loader is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:52859) ExperimentalWarning: --experimental-loader is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:52859) ExperimentalWarning: --experimental-loader is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:52859) ExperimentalWarning: --experimental-loader is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
  ✔ source › helpers › is-low-quality-comment › isLowQualityComment
  ✔ source › github-helpers › search-query › .get
  ✔ source › github-helpers › search-query › .getQueryParts
  ✔ source › github-helpers › search-query › getQueryParts with spaces support
  ✔ source › github-helpers › search-query › .set
  ✔ source › github-helpers › search-query › .edit
  ✔ source › github-helpers › search-query › .replace
  ✔ source › github-helpers › search-query › .remove
  ✔ source › github-helpers › search-query › .add
  ✔ source › github-helpers › search-query › .includes
  ✔ source › github-helpers › search-query › defaults
  ✔ source › github-helpers › search-query › deduplicate is:pr/issue
  ✔ source › github-helpers › search-query › remove additional spaces
  ✔ source › github-helpers › search-query › parse label link
  ✔ source › helpers › get-items-between › getItemsBetween
  ✔ source › helpers › loose-parse-int › looseParseInt
  ✔ source › github-helpers › parse-backticks › parseBackticks
  ✔ source › helpers › pluralize › pluralize
  ✔ source › helpers › select-has › basic :has() support (144ms)
  ✔ source › helpers › select-has › returns undefined if not found
  ✔ source › helpers › select-has › supports looking for descendants in base element
  ✔ source › helpers › select-has › supports looking for direct children in base element
  ✔ source › helpers › select-has › throws error when there’s a space before :has()
  ✔ source › helpers › select-has › throws error when there is more than one :has()
  ✔ source › helpers › select-has › throws on sibling selectors
  ✔ test › shorten-link › shorten link in comment text
  ✔ test › shorten-link › avoid shortening link in code block inside comment
  ✔ test › shorten-link › avoid shortening link in embedded file preview inside comment
  ✔ test › shorten-link › shorten link in review comment text
  ✔ test › shorten-link › avoid shortening links in suggestion inside review comment
  ✔ source › github-helpers › prevent-link-loss › preventPrCommitLinkLoss
  ✔ source › github-helpers › prevent-link-loss › preventPrCompareLinkLoss
  ✔ source › github-helpers › prevent-link-loss › preventDiscussionLinkLoss
  ✔ source › github-helpers › github-url › branch
  ✔ source › github-helpers › github-url › branch with multiple slashes
  ✔ source › github-helpers › github-url › object
  ✔ source › github-helpers › github-url › nested object
  ✔ source › github-helpers › github-url › change branch
  ✔ source › github-helpers › github-url › change filePath
  ✔ source › github-helpers › github-url › get filePath from search
  ✔ test › get-current-committish › getCurrentCommittish
  ✔ source › helpers › show-whitespace-on-line › showWhiteSpacesOnLine raw (358ms)
  ✔ source › helpers › show-whitespace-on-line › showWhiteSpacesOnLine real code (321ms)
  ✔ source › helpers › show-whitespace-on-line › showWhiteSpacesOnLine highlighted code (311ms)
  ✔ source › github-helpers › index › getConversationNumber (122ms)
  ✔ source › github-helpers › index › parseTag (105ms)
  ✔ source › github-helpers › index › compareNames
  ✔ source › github-helpers › index › getLatestVersionTag
  ✔ source › github-helpers › index › shouldFeatureRun
  ✔ source › github-helpers › index › addHotkey if one is specified
  ✔ source › github-helpers › index › addHotkey if the same is already specified
  ✔ source › github-helpers › index › addHotkey when none are specified
  ─

  52 tests passed
```

Screen Shot 11

# After
```
❯ npx vitest run

 RUN  v0.17.1 refined-github

 ✓ source/github-helpers/search-query.test.ts (13)
 ✓ source/helpers/select-has.test.ts (7)
 ✓ source/helpers/get-items-between.test.ts (1)
 ✓ source/helpers/loose-parse-int.test.ts (1)
 ✓ source/helpers/is-low-quality-comment.test.ts (1)
 ✓ source/helpers/pluralize.test.ts (1)
 ✓ source/github-helpers/parse-backticks.test.ts (1)
 ✓ source/helpers/show-whitespace-on-line.test.ts (3)
 ✓ source/github-helpers/github-url.test.ts (7)
 ✓ test/get-current-committish.test.ts (1)
 ✓ test/shorten-link.test.ts (5)
 ✓ source/github-helpers/index.test.ts (8)
 ✓ source/github-helpers/prevent-link-loss.test.ts (3)

Test Files  13 passed (13)
     Tests  52 passed (52)
      Time  2.28s (in thread 654ms, 349.02%)
```

Screen Shot 10

### CODE REVIEW

**Comment by yakov116** [comment]
Is it a federal crime to drop a sindre package?

**Comment by fregante** [comment]
> Is it a federal crime to drop a sindre package?

Shh don't tell him, he's AFK

**Comment by fregante** [comment]
I haven't even finished work on the `drop-xo` branch yet

File: source/github-helpers/index.test.ts [hunk 1]
```diff
@@ -74,110 +74,122 @@ test('getConversationNumber', t => {
...
-test('getLatestVersionTag', t => {
-	t.is(getLatestVersionTag([
+test('getLatestVersionTag', () => {
+	expect(getLatestVersionTag([
 		'0.0.0',
 		'v1.1',
 		'r2.0',
 		'3.0',
-	]), '3.0', 'Tags should be sorted by version');
+	])).toBe('3.0');
```
[the remark this answers is not in the document]
Comment (fregante) [reply source/github-helpers/index.test.ts]: Fixed!

**Pull request merged**